src/v037/server.rs

```diff
@@ -125,31 +199,77 @@ where
         ServerBuilder::default()
     }
 
-    pub async fn listen<A: ToSocketAddrs + std::fmt::Debug>(self, addr: A) -> Result<(), BoxError> {
-        tracing::info!(?addr, "starting ABCI server");
-        let listener = TcpListener::bind(addr).await?;
-        let local_addr = listener.local_addr()?;
-        tracing::info!(?local_addr, "bound tcp listener");
+    fn handle_new_connection(&self, socket: impl Into<SocketKind>) {
+        let socket = socket.into();
+        // set parent: None for the connection span, as it should
+        // exist independently of the listener's spans.
+        //let span = tracing::span!(parent: None, Level::ERROR, "abci", ?addr);
+        let conn = Connection {
+            consensus: self.consensus.clone(),
+            mempool: self.mempool.clone(),
+            info: self.info.clone(),
+            snapshot: self.snapshot.clone(),
+        };
+        //tokio::spawn(async move { conn.run(socket).await.unwrap() }.instrument(span));
```

Comment:
remove

Reply:
That's original tower-abci code just shuffled around. :-)